lua/copilot/extensions/auth.lua

```diff
@@ -1,5 +1,105 @@
+local util = require("copilot.util")
```

Comment:
Could you move auth to the parent dir? Extensions is primarily just extended copilot functionality

Reply:
You mean move the whole `lua/copilot/extensions/auth.lua` to `lua/copilot/auth.lua`?

Reply:
> You mean move the whole `lua/copilot/extensions/auth.lua` to `lua/copilot/auth.lua`?

Yeah. I think there should be a utils dir at some point, but for right now that is most consistent with how the project is structured and will make it easier to migrate to a directory dedicated to that later.

I know I had it in extensions, but that was mostly because it was an in progress thing and I planned to move it out once completed.